TypeSystem/flow-analysis/promotion_via_type_test_A04_t02.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via type test with type
+/// `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and `captured` is `false`
+/// - and `S` is the current type of `x` in `VM`
+/// - and not `S <: T`
+/// - and `T <: S` or (`S` is `X extends R` and `T <: R`) or (`S` is `X & R` and
+///   `T <: R`)
+///
+/// @description Checks that a variable is promoted after the type test in one
+/// of the branches, then is is not promoted in another branch and after it.
+/// Test type `void`.
```

Comment:
I'm afraid I'd recommend that this particular test is deleted.

The point is that a variable can never be promoted to have type `void` (because it would then have to have at least a declared type which is a proper supertype of `void`, and there is no such type). So we can't even get started investigating how a promotion to `void` would work, because they don't exist.